unison-core/src/Unison/NamesWithHistory.hs

```diff
@@ -274,7 +274,7 @@ suffixedTermName :: Int -> Referent -> NamesWithHistory -> [HQ'.HashQualified Na
                   isHQ'd = R.manyDom fqn rel -- it is conflicted
                   hq n = HQ'.take length (hq' n r)
                   hqn = if isHQ'd then hq n' else HQ'.fromName n'
-               in (isHQ'd, Name.countSegments fqn, Name.isAbsolute n', hqn)
+               in (isHQ'd, Name.countSegments n', Name.isAbsolute n', hqn)
```

Comment:
Prefer shortest name by the _abbreviated_ name, rather than the fully qualified name.